pxteditor/history.ts

```diff
@@ -375,6 +375,51 @@ namespace pxt.workspace {
         toWrite[pxt.HISTORY_FILE] = JSON.stringify(history);
     }
 
+    export function pushSnapshotOnHistory(text: ScriptText, currentTime: number) {
+        let history: pxt.workspace.HistoryFile;
```

Comment:
Not super necessary, and there might also be something that I'm missing, but have you considered putting this `let history .. if .. else.. ` chunk in a separate function to limit repeated code?

Reply:
it's slightly different in each function, sadly